core/state_transition.go

```diff
@@ -217,12 +217,12 @@ func IntrinsicGas(data []byte, accessList types.AccessList, isContractCreation b
 func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool, vmRunner vm.EVMRunner, sysCtx *SysContractCallCtx) *StateTransition {
 	var gasPriceMinimum *big.Int
 	if evm.ChainConfig().IsEspresso(evm.Context.BlockNumber) {
-		var feeCurrency *common.Address = msg.FeeCurrency()
 		if msg.MaxFeeInFeeCurrency() != nil {
 			// Celo denominated tx
-			feeCurrency = nil
+			gasPriceMinimum = sysCtx.GetGasPriceMinimum(nil)
+		} else {
+			gasPriceMinimum = sysCtx.GetGasPriceMinimum(msg.FeeCurrency())
 		}
-		gasPriceMinimum = sysCtx.GetGasPriceMinimum(feeCurrency)
```

Comment:
Why is this necessary? This seems to be functionally unchanged by the change. I.E `sysCtx.GetGasPriceMinimum` is called once with either nil or a fee currency.

Reply:
It is uncchanged. But since the bad block I wanted to make sure everything was like before, so while this is a diff here, it makes it closer than what we had before the cip66 change in state_transition. The line 224 is exactly what we had, I wanted to preserve that line, even if, functionally, this block is the same